=== modified file 'src/display.c'
Index: lightdm/src/display.c
===================================================================
--- lightdm.orig/src/display.c	2012-08-30 10:15:15.099230000 +1200
+++ lightdm/src/display.c	2012-08-30 13:50:46.157969462 +1200
@@ -623,6 +623,8 @@
     }
 }
 
+static void prepend_argv (gchar ***argv, const gchar *value);
+
 static gboolean
 display_start_greeter (Display *display)
 {
@@ -650,6 +652,8 @@
     if (!argv)
         return TRUE;
 
+    prepend_argv (&argv, "/usr/lib/lightdm/lightdm-greeter-session");
+
     session_run (display->priv->session, argv);
 
     return FALSE;
