diff -Nru defer-1.0.6.orig/defer/__init__.py defer-1.0.6/defer/__init__.py
--- defer-1.0.6.orig/defer/__init__.py	2012-09-25 00:24:34.513300646 -0400
+++ defer-1.0.6/defer/__init__.py	2012-09-25 00:25:04.990284982 -0400
@@ -75,8 +75,6 @@
 
 __version__ = version.VERSION
 
-PY3K = sys.version_info.major > 2
-
 
 class _DefGen_Return(BaseException):
     """Exception to return a result from an inline callback."""
@@ -126,10 +124,7 @@
 
     def raise_exception(self):
         """Raise the stored exception."""
-        if PY3K:
-            raise self.value.with_traceback(self.traceback)
-        else:
-            raise self.type, self.value, self.traceback
+        raise self.value.with_traceback(self.traceback)
 
     def catch(self, *errors):
         """Check if the stored exception is a subclass of one of the
@@ -478,13 +473,8 @@
             # Send the last result back as the result of the yield expression.
             is_failure = isinstance(result, DeferredException)
             if is_failure:
-                if PY3K:
-                    excep = result.value.with_traceback(result.traceback)
-                    result = gen.throw(excep)
-                else:
-                    result = gen.throw(result.type,
-                                       result.value,
-                                       result.traceback)
+                excep = result.value.with_traceback(result.traceback)
+                result = gen.throw(excep)
             else:
                 result = gen.send(result)
         except StopIteration:
